source/read-write-concern/read-write-concern.rst

```diff
@@ -380,6 +381,38 @@ Drivers SHOULD NOT parse server replies for "writeConcernError" in generic comma
 (Reporting of writeConcernErrors is more complex for bulk operations,
 see the Bulk API Spec.)
 
+writeConcernError Examples
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The set of possible writeConcernErrors is quite large because they can include
+errors caused by shutdown, stepdown, interruption, maxTimeMS, and wtimeout.
+This section attempts to list all known error codes that may appear
+within a writeConcernError but may not be exhaustive. Note that some errors
+have been abbreviated:
+
+- ``{ok:1, writeConcernError: {code: 91, codeName: "ShutdownInProgress"}}``
+- ``{ok:1, writeConcernError: {code: 189, codeName: "PrimarySteppedDown"}}``
+- ``{ok:1, writeConcernError: {code: 11600, codeName: "InterruptedAtShutdown"}}``
+- ``{ok:1, writeConcernError: {code: 11601, codeName: "Interrupted"}}``
+- ``{ok:1, writeConcernError: {code: 11602, codeName: "InterruptedDueToReplStateChange"}}``
+- ``{ok:1, writeConcernError: {code: 64, codeName: "WriteConcernFailed", errmsg: "waiting for replication timed out", errInfo: {wtimeout: True}}}``
+- ``{ok:1, writeConcernError: {code: 64, codeName: "WriteConcernFailed", errmsg: "multiple errors reported : {...} at shardName1 :: and :: {...} at shardName2"}}`` [#]_
+- ``{ok:1, writeConcernError: {code: 50, codeName: "MaxTimeMSExpired"}}``
+- ``{ok:1, writeConcernError: {code: 100, codeName: "UnsatisfiableWriteConcern", errmsg: "Not enough data-bearing nodes"}}``
+- ``{ok:1, writeConcernError: {code: 79, codeName: "UnknownReplWriteConcern"}}``
+
+Note also that it is possible for a writeConcernError to be attached to a
+command failure. For example:
+
+- ``{ok:0, code: 251, codeName: "NoSuchTransaction", writeConcernError: {code: 91, codeName: "ShutdownInProgress"}}`` [#]_
+
+.. [#] This is only possible in a sharded cluster. When a write goes to
+       multiple shards and more than one shard returns a writeConcernError,
+       then mongos will construct a "WriteConcernFailed" error where the errmsg
+       field contains the stringified writeConcernError from each shard.
```

Comment:
Can you be explicit here that different shards can return different write concern errors so this error message can refer to multiple different errors?

Reply:
The footnote says that this pertains to a sharded cluster where WC errors may be concatenated, but the example doesn't show that. Do we have a better example to demonstrate the concatenation behavior?

Specifically, I'm curious what the numeric `code` field will look like. I don't believe there's a code in [error_codes.err](https://github.com/mongodb/mongo/blob/master/src/mongo/base/error_codes.err) specific to a mongos merge scenario.